contracts/staking/PoSValidatorManager.sol

```diff
@@ -129,25 +139,44 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         uint32 messageIndex
     ) external {
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+
+        Validator memory validator = _initializeEndValidation(validationID);
+
         // Check that minimum stake duration has passed
-        Validator memory validator = getValidator(validationID);
         require(
-            block.timestamp
+            validator.endedAt
                 >= validator.startedAt + $._validatorRequirements[validationID].minStakeDuration,
             "PoSValidatorManager: minimum stake duration not met"
         );
 
         if (includeUptimeProof) {
-            _getUptime(validationID, messageIndex);
+            uint64 uptime = _getUptime(validationID, messageIndex);
```

Comment:
can you add comments here explaining the value from uptime, and what the check is checking